### ISSUE TITLE
draw_bounding_boxes should check that (W, H) >= (xmax, ymax) >= (xmin, ymin) >= (0, 0)

### ISSUE THREAD

**Issue by norabelrose** [pytorch/vision#5517]
### 🚀 The feature

Currently, `draw_bounding_boxes` does not sanity check the input bounding boxes to ensure that `(xmax, ymax) >= (xmin, ymin)`. This can lead to unexpected output that can be difficult to debug if the user, say, inputs boxes in cxcywh format.

I could make a PR for this if needed but would prefer if someone else did it since I have my hands full with a PR I'm working on for PyTorch proper.

### Motivation, pitch

I personally just ran into this problem and I thought something was wrong with my network for a while.

Currently, the box conversion ops do this check, and it seems at least as important to do the same check in `draw_bounding_boxes`.

**Comment by pmeier** [comment]
I agree with the overall sentiment here. My question is how you got to the situation. Did you know that you had "cxcywh" bounding boxes and missed that `draw_bounding_boxes` requires "xyxy" or did you not know the format before and only started to look into it after you got wrong results?

In the former case, we also could have a `format="xyxy"` parameter on the function, since we know how to convert bounding boxes. In general though, a smoke test after the conversion is still something we could do.

We are currently in the process of introducing new abstractions for bounding boxes. With that, bounding boxes store information about the format and thus a conversion as proposed above would be handled automatically.

https://github.com/pytorch/vision/blob/71d2bb0bc67044f55d38bfddf04e05be0343deab/torchvision/prototype/features/_image.py#L83-L86

In an internal discussion following https://github.com/pytorch/vision/pull/5500#discussion_r816501046, we decided to move this functionality into `torchvision.utils` just before we migrate the new abstractions to the main area.

**Comment by norabelrose** [comment]
@pmeier I knew that `draw_bounding_boxes` requires "xyxy", but I got confused about which format my bounding boxes were at that particular point in the code, since I'm using a DETR model where cxcywh is used for the L1 loss but xyxy is used for the GIOU.

The class abstraction definitely sounds like a good idea. I still think though that it would be a good idea to add this check to the "functional" version `draw_bounding_box` API, especially since the conversion ops already do it.

**Comment by NicolasHug** [comment]
@norabelrose thanks for the suggestion, we'd be happy to review a PR that adds these checks

**Comment by oke-aditya** [comment]
To avoid code duplication and for convenience.

I suggest we create a small util _check_box_fmt. We need these is multiple places. `box_iou`, `generalized_box_iou`, `nms` and `draw_bounding_boxes`.

With this single util we could check all places and raise ValueError suggesting to either check the input or use `box_convert` to change the format. 

cc @NicolasHug @norabelrose 
Let me know if you didn't have bandwidth / need help.

**Comment by NicolasHug** [comment]
There was some discussions before about adding these checks in the ops (box_iou, nms etc.). We don't do it because this would require adding a synchronization point of the CUDA streams. It's probably best to ignore the ops for now and stick to `draw_bounding_boxes`

**Comment by singhularAdi** [comment]
I could look into this if no one else is.

**Comment by pmeier** [comment]
Sure, go ahead!

**Comment by oke-aditya** [comment]
Go ahead @singhularAdi Feel free to ping Nicolas for review :) 
Note that you should throw a ValueError if the above case occurs
